test/test.js

```diff
@@ -106,16 +152,284 @@ test('esLoader module', function(t){
     t.ok(isStream.isWritable(loader), 'esLoader.load returns a write stream');
   });
 
+  esLoader.loadIntoIndex({client: ''}, function(err){
+    t.ok(err, 'Loader errors without proper arguments');
+  });
+
+  var op2 = JSON.parse(JSON.stringify(options));
+  op2.client = client;
+  op2.forcedIndex = 'testforcedindex';
+
+  esLoader.loadIntoIndex(op2, function(err, loader){
+    t.notOk(err, 'Proper arguments to loader doesn\'t error');
+    t.ok(isStream.isWritable(loader), 'esLoader.load returns a write stream');
+  });
+
+});
+
+
+
+
+test('createIndex module', function(t){
+  t.plan(3);
+
+  var client = options.client;
+
+  createIndex({alias: 'testalias', client: client}, 'testname', function(err, index){
+    if(err) t.fail();
+    t.ok(index, 'Assigns index for arbitrary alias.');
+  });
+
+  createIndex({alias: 'census', client: client}, 'testname', function(err, index){
+    if(err) t.fail();
+    t.ok(index, 'Creates index for census.');
+    client.indices.delete({index: index}, function(err){
+      t.notOk(err, 'Deletes test index');
+    });
+  });
+});
+
+
+
+
+test('bulk Prefixer', function(t){
+  t.plan(3);
+
+  t.ok(isStream.isDuplex(bulkPrefixer()), 'bulkPrefixer returns a stream');
+
+  var pref = bulkPrefixer();
+  var pref2 = bulkPrefixer();
+  var prefixed = '{"index":{}}\nsomedata\n';
+
+  pref.end('somedata\n');
+  pref2.end('somedata');
+
+  pref.on('data', function(data){
+    t.equal(data.toString(), prefixed, 'Properly prefixes newline-delimited data');
+  });
+
+  pref2.on('data', function(data){
+    t.equal(data.toString(), prefixed, 'Properly prefixes data with missing newline');
+  });
+});
+
+
+
+
+test('handleCsv module', function(t){
+  t.plan(6);
+  var csvFile = 'test/data/virginia.csv';
+  var txtFile = 'test/data/virginia.txt';
+  var csvStream = fs.createReadStream(csvFile);
+  var txtStream = fs.createReadStream(txtFile);
+  var txtRecord = {name: 'virginia', file: 'virginia.txt', spatialReference: 'NAD83'}
+  var csvRecord = {name: 'virginia', file: 'virginia.csv', spatialReference: 'NAD83'}
+  var badTxtRecord = {name: 'virginia', file: 'virginia.txt'};
+
+  handleCsv(csvFile, csvRecord, scratchSpace, function(record, vrt){
+    t.ok(vrt, 'Creates a valid vrt file from a csv file and good record.');
+  },
+  function(record, err){
+   if(err) t.fail('Unexpected error.');
+  });
+
+  handleCsv(csvStream, csvRecord, scratchSpace, function(record, vrt){
+    t.ok(vrt, 'Creates a valid vrt file from a csv stream and good record.');
+  },
+  function(record, err){
+   if(err) t.fail('Unexpected error.');
+  });
+
+  handleCsv(txtFile, txtRecord, scratchSpace, function(record, vrt){
+    t.ok(vrt, 'Creates a valid vrt file from a text file and good record.');
+    fs.copySync('test/data/virginia.csv', 'test/data/virginia.txt');
```

Comment:
ogr2ogr will not function appropriately on csv files unless they have the .csv extension.